src/md/release.md

```diff
@@ -150,6 +150,47 @@ mkdir svn-dist && cp *-source-* svn-dist/
 svn import svn-dist https://dist.apache.org/repos/dist/dev/apex/${RNAME}-RC1 -m "Apache Apex v${rv}-RC1"
 ```
 
+## Build and Deploy Documentation
+
+The documentation will be generated as static HTML files and copied into the ```apex-site``` repository. It will be available at an alternate URL location until the release is promoted, at which time it will also be promoted to the main website location.
+
+Do the following setup setups before building and deploying the documentation. Clone the apex-site repository into a folder called ```apex-site``` at the same level as the current repository. Set the following environment variables
+
+For -core releases:
+
+```
+REPO_NAME=apex-core
+DOC_NAME=apex
+```
+
+For -malhar releases:
+
+```
+REPO_NAME=apex-malhar
+DOC_NAME=malhar
+```
+
+Execute the following commands to build and deploy the documentation. **Note**: Until [mkdocs #859](https://github.com/mkdocs/mkdocs/issues/859) is resolved and available for download, use mkdocs built against [master](https://github.com/mkdocs/mkdocs).
+
+```bash
+# build docs, they would be generated in the site folder
+mkdocs build --clean
+
+# copy docs from site into target folder on apex-site
+cd ../apex-site
+git checkout asf-site
+rm -rf docs/${DOC_NAME}-${rv}
```

Comment:
@PramodSSImmaneni I see no need for staging documentation. The documentation can be published under the final name right away. Nobody is going to look at it before the release is out, as that's what establishes the link to the docs.

For the case of updating latest (different from your example), that can be done anytime, and maybe it should not be pointed to a X.Y version before it was released. But it could theoretically also be pointed to latest master version, I don't think we had a need for that so far, as long as we release frequently. And it is possible to look at individual pages in github anyways.